infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -417,6 +429,49 @@ protected void initChannel(@Nonnull Channel ch) throws Exception {
         };
     }
 
+    /**
+     * Cleanly shuts down the server and restarts.
+     *
+     * @param serverContext Server Context.
+     * @param resetData     Resets and clears all data if True.
+     */
+    public static void restartServer(ServerContext serverContext, boolean resetData) {
+
+        final Thread previousServerThread = corfuServerThread;
+        final Map<String, Object> opts = serverContext.getServerConfig();
+
+        corfuServerThread = new Thread(() -> {
+            cleanShutdown((NettyServerRouter) serverContext.getServerRouter());
+            if (resetData && !(Boolean) serverContext.getServerConfig().get("--memory")) {
+                File serviceDir = new File((String) serverContext.getServerConfig()
+                        .get("--log-path"));
+                try {
+                    FileUtils.cleanDirectory(serviceDir);
+                } catch (IOException ioe) {
+                    throw new UnrecoverableCorfuError(ioe);
+                }
+            }
+            serverContext.close();
+
+            // Wait for previous server thread to join.
+            try {
+                previousServerThread.join();
+                Runtime.getRuntime().removeShutdownHook(shutdownThread);
+            } catch (InterruptedException ie) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)